src/components/z-popover/index.tsx

```diff
@@ -0,0 +1,151 @@
+import { Component, Prop, h, Host, State, Listen } from "@stencil/core";
+import classNames from "classnames";
+import {
+  PopoverPosition,
+  PopoverBorderRadius,
+  PopoverShadow,
+  KeyboardKeys,
+} from "../../beans";
+import { getElementTree } from "../../utils/utils";
+
+@Component({
+  tag: "z-popover",
+  styleUrl: "styles.css",
+  shadow: false,
+  scoped: true,
+})
+export class ZPopover {
+  /** [optional] Popover position */
+  @Prop({ mutable: true }) position?: PopoverPosition =
+    PopoverPosition["after-up"];
+  /** [optional] Background color token for popover */
+  @Prop() backgroundColor?: string = "color-white";
+  /** [optional] Border radius token for popover */
+  @Prop() borderRadius?: PopoverBorderRadius = PopoverBorderRadius.small;
+  /** [optional] Box shadow token for popover */
+  @Prop() boxShadow?: PopoverShadow = PopoverShadow["shadow-1"];
+  /** [optional] Show or hide arrow */
+  @Prop() showArrow?: boolean = false;
+
+  @State() isVisible: boolean = false;
+
+  private popoverElem: HTMLElement;
+  private defaultPosition: PopoverPosition = this.position;
+
+  @Listen("closePopover")
+  closePopover() {
+    this.isVisible = false;
+  }
+
+  @Listen("keyup", { target: "window" })
+  closePopoverWithKeyboard(e: any) {
+    if (e.key === KeyboardKeys.ESC) {
+      this.closePopover();
+    }
+  }
+
+  handleClick() {
+    this.isVisible = !this.isVisible;
+    this.checkSpaceAvailable();
+  }
+
+  @Listen("click", { target: "body", capture: true })
+  handleOutsideClick() {
```

Comment:
The click event on the popover itself or on a non-interactive element inside it (paragraph, label , empty div etc) closes the popover.
I don't think this behaviour is desired and  it also causes issues during text selection inside the element